### PR TITLE
Add midiDevices to query available devices

### PR DESCRIPTION
**Pull request by TristanCacqueray**
This change adds a new binding to the FRP.Event.MIDI module and
updates the WTK example to display the available devices list.

### CODE REVIEW

**Comment by mikesol** [comment]
This is really nice @TristanCacqueray ! I haven't tested out the WTK example in a while - have you tested it to make sure the example still works? If so I'll merge it, otherwise we can debug. Thanks so much for putting this together!

**Comment by TristanCacqueray** [comment]
Yes, the example still works and it shows the available devices. Though I'll tweak the addition to split the midiDevices accessor in two for IN and OUT, since their usage are really different.

Then if you don't mind, I think we need to change the `midi` function to take a list of device instead of grabbing the input from all sources.

File: src/FRP/Event/MIDI.purs [hunk 1]
```diff
@@ -90,6 +115,16 @@ toMIDIEvent =
 fromEvent :: WE.Event -> Maybe MIDIMessageEvent
 fromEvent = unsafeReadProtoTagged "MIDIMessageEvent"
 
+midiInputDevices :: MIDIAccess -> Effect (List MIDIDevice)
+midiInputDevices midiAccess_ = do
+  devices <- midiInputDevices_ midiAccess_ mkMIDIDevice
+  pure $ fromFoldable devices
```
Comment (mikesol) [inline src/FRP/Event/MIDI.purs:121]:
```suggestion
midiInputDevices midiAccess_ =
  fromFoldable <$> midiInputDevices_ midiAccess_ mkMIDIDevice
```

File: src/FRP/Event/MIDI.purs [hunk 2]
```diff
@@ -90,6 +115,16 @@ toMIDIEvent =
...
+
+midiOutputDevices :: MIDIAccess -> Effect (List MIDIDevice)
+midiOutputDevices midiAccess_ = do
+  devices <- midiOutputDevices_ midiAccess_ mkMIDIDevice
+  pure $ fromFoldable devices
```
Comment (mikesol) [inline src/FRP/Event/MIDI.purs:126]:
```suggestion
midiOutputDevices midiAccess_ =
  fromFoldable <$> midiOutputDevices_ midiAccess_ mkMIDIDevice
```

**Review by mikesol — APPROVED** [review]
Approved! I have two minor suggestions, neither of which are necessary. They're minor style things. When possible & when I remember, I try to use a minimal subset of typeclass instances, so instead of `Bind` and `Applicative` from `Effect` one can just use `Functor` here. But it's a minor tweak, up to you!

You can either incorporate the tweaks or leave as is and then, barring no objections, I'll merge tomorrow morning (unless you need it in `main` earlier). Thank you for your work on this!

**Comment by TristanCacqueray** [comment]
Alright, thank you for the suggestion, it makes sense. No hurry, I'm just getting familiar with the code.

**Comment by mikesol** [comment]
Thanks for the PR 🚀

**Comment by TristanCacqueray** [comment]
You are welcome, thanks for the help!

**Pull request merged**